discovery.md

```diff
@@ -0,0 +1,436 @@
+  - "GitHub.com"
+  - "Awesome Cloud Storage"
+  - "com.example.microservices.userlogin"
+
+#### produceruri
+
+- Type: `URI`
+- Description: Absolute URI that provides a link back to the producer, or
+  documentation about the producer. This is for human consumption.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty URI
+- Examples:
+  - "http://cloud.example.com/docs/blobstorage"
+
+#### source
+
+- Type: `List` of `String`
+- Description: CloudEvents [`source`](https://github.com/cloudevents/spec/blob/master/spec.md#source-1)
+  attribute values available from this producer for this event type. Values here
+  may be used to create subscriptions from this producer.
+- Constraints
+  - OPTIONAL
+- Examples:
+  - Single Internet-wide unique URI with a DNS authority.
+   - ["https://github.com/cloudevents"]
+  - Application-specific identifiers
+   - ["/sensors/tn-1234567/alerts", "/sensors/tr-1234567/alerts"]
+
+#### sourcestructure
+
+- Type: `String`
+- Description: Documentation of the format of the source attribute for this
+  producer and event type combination.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty string
+- Examples:
+  - "github.com/[organization]/[repository]/pull/[id]"
+
+#### type
+
+- Type: `String`
+- Description: CloudEvents [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
+  attribute.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string, following the constraints as defined in the
+    CloudEvents spec.
+- Examples:
+  - "com.github.pull.create"
+  - "com.example.object.delete.v2"
+
+#### specversion
+
+- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Description: CloudEvents [`specversion`](https://github.com/cloudevents/spec/blob/master/spec.md#specversion)
+  that will be used for events published for this producer, event type
+  combination. Compliant event producers MUST use a value of `1.0` when
+  referring to this version of the specification.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+
+#### datacontenttype
+
+- Type: `String`
+- Description: CloudEevnts [`datacontenttype`](https://github.com/cloudevents/spec/blob/master/spec.md#datacontenttype)
+  attribute. Indicating how the `data` attribute of subscribed events will be
+  encoded.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST adhere to the format specified in [RFC 2046](https://tools.ietf.org/html/rfc2046)
+
+#### dataschema
+
+- Type: `URI`
+- Description: CloudEevnts [`datacontenttype`](https://github.com/cloudevents/spec/blob/master/spec.md#dataschema)
+  attribute. This identifies the canonical storage location of the schema of
+  the `data` attribute of subscribed events.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty URI
+
+#### dataschemacontent
+
+- Type: `String`
+- Description: An inline representation of the schema of the `data` attribute
+  encoding mechanism. This is an alternative to using the `dataschema`
```

Comment:
We may need to define the language/spec used

Reply:
added "dataschematype"